src/Cake.Testing/Extensions/FakeFileExtensions.cs

```diff
@@ -44,12 +46,38 @@ public static FakeFile SetContent(this FakeFile file, string content)
         }
 
         /// <summary>
-        /// Gets the text content of the file (UTF-8 encoding).
+        /// Gets the binary content of the specified file.
         /// </summary>
         /// <param name="file">The file.</param>
+        /// <returns>The binary content of the specified file.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times")]
+        public static byte[] GetBinaryContent(this FakeFile file)
+        {
+            if (file == null)
+            {
+                throw new ArgumentNullException(nameof(file));
+            }
+            if (!file.Exists)
+            {
+                throw new FileNotFoundException("File could not be found.", file.Path.FullPath);
+            }
+            using (var stream = file.OpenRead())
+            using (var reader = new BinaryReader(stream))
+            using (var memory = new MemoryStream())
+            {
+                reader.BaseStream.CopyTo(memory);
+                return memory.ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Gets the text content of the file.
+        /// </summary>
+        /// <param name="file">The file.</param>
+        /// <param name="encoding">The text encoding. Defaults to UTF8 without BOM.</param>
         /// <returns>The text content of the file.</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times")]
-        public static string GetTextContent(this FakeFile file)
+        public static string GetTextContent(this FakeFile file, Encoding encoding = null)
```

Comment:
To not break ABI add an `GetTextContent(this FakeFile file)` overload calling the new one instead of default parameter.